source_gen/lib/src/generator_for_annotation.dart

```diff
@@ -123,4 +138,27 @@ abstract class GeneratorForAnnotation<T> extends Generator {
     ConstantReader annotation,
     BuildStep buildStep,
   ) {}
+
+  /// Implement to return source code to generate for [directive].
+  ///
+  /// This method is invoked based on finding directives annotated with an
+  /// instance of [T]. The [annotation] is provided as a [ConstantReader].
+  ///
+  /// Supported return values include a single [String] or multiple [String]
+  /// instances within an [Iterable] or [Stream]. It is also valid to return a
+  /// [Future] of [String], [Iterable], or [Stream]. When multiple values are
+  /// returned through an iterable or stream they will be deduplicated.
+  /// Typically each value will be an independent unit of code and the
+  /// deduplication prevents re-defining the same member multiple times. For
+  /// example if multiple annotated elements may need a specific utility method
+  /// available it can be output for each one, and the single deduplicated
+  /// definition can be shared.
+  ///
+  /// Implementations should return `null` when no content is generated. Empty
+  /// or whitespace-only [String] instances are also ignored.
+  dynamic generateForAnnotatedDirective(
```

Comment:
What's the deal w/ this @scheglov ?

Reply:
I'm sorry, i don't understand the question.
This method is invoked at the line `62`.

We cannot use the same method for elements and directive.

Reply:
I think we can skip this for now if `checks` is the only known usage that needs it. I can add it later if it becomes necessary.

We will need to be careful not to break internal uses of this and document carefully the impact when we publish externally since this is fairly subtle and it isn't clear in source code exactly when it will matter.